config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -416,6 +440,432 @@ type MetricsServerConfig struct {
+	Name *string `json:"name,omitempty"`
+	// remoteTimeout is the timeout for requests to the remote write endpoint.
+	// When omitted, the default is 30s.
+	// +optional
+	// +kubebuilder:validation:MaxLength=20
+	// +kubebuilder:validation:MinLength=2
+	RemoteTimeout string `json:"remoteTimeout,omitempty"`
+	// writeRelabelConfigs is a list of relabeling rules to apply before sending data to the remote endpoint.
+	// Maximum of 10 relabeling rules can be specified.
+	// +optional
+	// +kubebuilder:validation:MaxItems=10
+	// +listType=atomic
+	WriteRelabelConfigs []RelabelConfig `json:"writeRelabelConfigs,omitempty"`
+}
+
+// RelabelConfig represents a relabeling rule.
+type RelabelConfig struct {
+	// sourceLabels specifies which labels to extract from each series for this relabeling rule.
+	// If a label does not exist, an empty string ("") is used in its place.
+	// The values of these labels are joined together using the configured separator,
+	// and the resulting string is then matched against the regular expression for
+	// the replace, keep, or drop actions.
+	// +optional
+	// +kubebuilder:validation:MaxItems=10
+	// +kubebuilder:validation:items:MaxLength=63
+	// +listType=set
+	SourceLabels []string `json:"sourceLabels,omitempty"`
+	// separator is the separator used to join source label values.
+	// +optional
+	// +kubebuilder:validation:MaxLength=10
+	Separator *string `json:"separator,omitempty"`
+	// regex is the regular expression to match against the concatenated source label values.
+	// +optional
+	// +kubebuilder:validation:MaxLength=1000
+	Regex *string `json:"regex,omitempty"`
+	// targetLabel is the target label name.
+	// +optional
+	// +kubebuilder:validation:MaxLength=63
+	TargetLabel *string `json:"targetLabel,omitempty"`
+	// replacement is the value against which a regex replace is performed if the
+	// regular expression matches. Regex capture groups are available.
+	// +optional
+	// +kubebuilder:validation:MaxLength=255
+	Replacement *string `json:"replacement,omitempty"`
+	// action is the action to perform.
+	// Valid actions are: replace, keep, drop, hashmod, labelmap, labeldrop, labelkeep.
+	// +required
+	Action RelabelAction `json:"action,omitempty"`
+}
+
+// TLSConfig represents TLS configuration for Alertmanager connections.
+type TLSConfig struct {
+	// ca is the CA certificate to use for TLS connections.
+	// +optional
+	CA SecretKeySelector `json:"ca,omitempty,omitzero"`
+	// cert is the client certificate to use for TLS connections.
+	// +optional
+	Cert SecretKeySelector `json:"cert,omitempty,omitzero"`
+	// key is the client key to use for TLS connections.
+	// +optional
+	Key SecretKeySelector `json:"key,omitempty,omitzero"`
+	// serverName is the server name to use for TLS connections.
+	// If specified, must be a valid DNS subdomain as per RFC 1123.
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	ServerName string `json:"serverName,omitempty"`
```

Comment:
What does it mean to omit these values?

Additionally, explicitly mention in the GoDoc that these are optional fields.